docs/UserGuide.md

```diff
@@ -41,146 +31,253 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-> The index **must be a positive integer** 1, 2, 3, ...
+### 2.7. Exiting the program : `exit`
+
+Exits the program.<br>
+Format: `exit`
+
+### 2.8. Undo a command : `undo`
+
+Undoes previous commands by the user.<br>
+Format: `undo [NUMBER]`
+
+> Undoes the last data-mutating command inputted by the user.<br>
+> If a number is entered, undoes that ammount of previous commands instead.<br>
+> The number **must be a positive integer** 1, 2, 3, ...
 
 Examples:
 
-* `list`<br>
-  `select 2`<br>
-  Selects the 2nd person in the address book.
-* `find Betsy` <br>
-  `select 1`<br>
-  Selects the 1st person in the results of the `find` command.
+* `add Test`<br>
+  `undo`<br>
+  Undoes adding Test to the todo list.
+* `add Assignment` <br>
+  `add Project` <br>
+  `undo 2`<br>
+  Undoes both commands.
 
-### 2.8. Clearing all entries : `clear`
+### 2.9. Redo a command : `redo`
 
-Clears all entries from the address book.<br>
-Format: `clear`
+Redoes previously undone commands by the user.<br>
+Format: `redo [NUMBER]`
 
-### 2.9. Exiting the program : `exit`
+> Redoes the last data-mutating command inputted since the undone point.<br>
+> If a number is entered, redoes that ammount of previous commands instead.<br>
+> The number **must be a positive integer** 1, 2, 3, ...
+> The number must be less than or equal to the number of commands undone.
 
-Exits the program.<br>
-Format: `exit`
+Examples:
+
+* `add Test`<br>
+  `undo`<br>
+  `redo`
+  Redoes adding Test to the todo list.
+* `add Assignment` <br>
+  `add Project`<br>
+  `undo 2`<br>
+  `redo`<br>
+  Redoes `add Assignment`.
 
 ### 2.10. Saving the data
 
-Address book data are saved in the hard disk automatically after any command that changes the data.<br>
+Todo list data are saved in the hard disk automatically after any command that changes the data.<br>
 There is no need to save manually.
 
-## 3. FAQ
+### 2.11. Viewing previous commands and accessing them : `history`
 
-**Q**: How do I transfer my data to another Computer?<br>
-**A**: Install the app in the other computer and overwrite the empty data file it creates with
-       the file that contains the data of your previous Address Book folder.
+Shows previous commands entered
+Format: `history`
 
-## 4. Command Summary
+> * Previous commands are listed in order from latest command to earlier command
+> * Alternatively, pressing on the <kbd>up</kbd> and <kbd>down</kbd> arrow keys on the keyboard will cycle through the commands previously entered
+
+Examples:
+* `add Test` <br>
+  `history` <br>
+  Shows `add Test` in the list.
+* `add Test` <br>
+  you press on the <kbd>up</kbd>kbd> arrow key. <br>
+  Shows `add Test` in your input text field.
 
-* **Add**  `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` <br>
-  e.g. `add James Ho p/22224444 e/jamesho@gmail.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+### 2.12. Add alias for commands: `add alias`
+
```

Comment:
@Rinder5 refer to the discussion over here https://github.com/CS2103JAN2017-W13-B3/main/issues/17